src/main/java/de/tum/cit/aet/artemis/iris/web/IrisResource.java

```diff
@@ -47,4 +74,66 @@ public ResponseEntity<IrisStatusDTO> getStatus() {
         return ResponseEntity.ok(new IrisStatusDTO(health.getStatus() == Status.UP, rateLimitInfo));
     }
 
+    /**
+     * Retrieves the overall ingestion state of a lecture by communicating with Pyris.
+     *
+     * <p>
+     * This method sends a GET request to the external Pyris service to fetch the current ingestion
+     * state of all lectures in a course, identified by its `lectureId`. The ingestion state can be aggregated from
+     * multiple lecture units or can reflect the overall status of the lecture ingestion process.
+     * </p>
+     *
+     * @param courseId the ID of the lecture for which the ingestion state is being requested
+     * @return a {@link ResponseEntity} containing the {@link IngestionState} of the lecture,
+     */
+    @GetMapping("courses/{courseId}/lectures/ingestion-state")
+    @EnforceAtLeastInstructorInCourse
+    public ResponseEntity<Map<Long, IngestionState>> getStatusOfLectureIngestion(@PathVariable long courseId) {
+        try {
+            Course course = courseRepository.findByIdElseThrow(courseId);
+            authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+            return ResponseEntity.ok(pyrisWebhookService.getLecturesIngestionState(courseId));
+        }
+        catch (EntityNotFoundException e) {
+            log.error("Error finding course while fetching ingestion state: ", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
```

Comment:
You translate a 404 Not Found to a 500 Internal Server Error. I don't think this is good. I suggest to move

Course course = courseRepository.findByIdElseThrow(courseId);

before the try block, because you typically do not want to catch an EntityNotFoundException